examples/mnist_test_oss.py

```diff
@@ -0,0 +1,169 @@
+## adapted from https://github.com/pytorch/examples/blob/master/mnist/main.py
+from __future__ import print_function
+import argparse
+import torch
+import torch.distributed as dist
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.optim as optim
+from torchvision import datasets, transforms
+from torch.optim.lr_scheduler import StepLR
+
+import time
+from fairscale.optim.oss import OSS
+import torch.multiprocessing as mp
+from fairscale.nn.data_parallel import ShardedDataParallel
+
+# how is world_size defined?
+WORLD_SIZE = 2
+OPTIM = torch.optim.RMSprop
+BACKEND = dist.Backend.NCCL if torch.cuda.is_available() else dist.Backend.GLOO  # type: ignore
+
+def dist_init(rank, world_size, backend):
+    print(f"Using backend: {backend}")
+    dist.init_process_group(backend=backend, init_method="tcp://localhost:29501", rank=rank, world_size=world_size)
+
+class Net(nn.Module):
```

Comment:
to me this adds information (one could wonder whether it's there because the model needs to be defined in a given way for instance), and it happens to be irrelevant information. That's why the other tutorial was using a predefined model from torchvision, to tentatively show that the model was not relevant